src/PolykeyAgent.ts

```diff
@@ -575,9 +620,9 @@ class PolykeyAgent {
         ingressPort: networkConfig_.ingressPort,
         tlsConfig,
       });
-      await this.nodeManager.start({ fresh });
-      await this.nodeManager.getConnectionsToSeedNodes();
-      await this.nodeManager.syncNodeGraph();
+      await this.nodeConnectionManager.start();
+      await this.nodeGraph.start({ fresh });
+      await this.nodeConnectionManager.syncNodeGraph();
```

Comment:
Now with the new architecture, does the synchronisation of the node graph have to be a separate step now? Can it be built into `nodeConnectionManager.start()` or `nodeGraph.start()`?

Reply:
I'd assume that we can integrate this into one of those start functions.

@tegefaulkes take a look at https://github.com/MatrixAI/js-polykey/pull/289#issuecomment-987412188 if you're not sure what this refers to. Otherwise, we can have a chat about this.

Reply:
Made it so it's called by `NodeConnectionManager.start()`.

Reply:
We can't sync the node Graph in the `start` method since it depends on some public methods. I'm going to revert the changes for now.

Reply:
So when this happened before, I resolved this by factoring out common code and making it a protected function that is used by both the public method and the start method. However if the code is simple, it's easy to just copy paste the functionality, DRY is not an ironclad rule.

Reply:
I'd need a deeper explanation about this. Wednesday we're both in, so let's get over it @joshuakarp @tegefaulkes.

Reply:
Ok for now this is fine. A longer term fix would be in `js-async-init` where we allow `ready` to have additional specification about it being callable while `starting` or `stopping`... etc.

Reply:
I've created a new issue here: https://github.com/MatrixAI/js-async-init/issues/7 to address the flexibility of `ready` which would allow the usage of `syncNodeGraph` inside the `ready` method.